src/HtsSystemContractJson.sol

```diff
@@ -66,6 +67,479 @@ contract HtsSystemContractJson is HtsSystemContract {
+        slot = bytes32(uint256(slot) + 1);
+
+        // Token supply type
+        storeBool(address(this), uint256(slot), tokenInfo.token.tokenSupplyType);
+        slot = bytes32(uint256(slot) + 1);
+
+        // Max supply
+        storeInt(address(this), uint256(slot), tokenInfo.token.maxSupply);
+        slot = bytes32(uint256(slot) + 1);
+
+        // Freeze default
+        storeBool(address(this), uint256(slot), tokenInfo.token.freezeDefault);
+        slot = bytes32(uint256(slot) + 1);
+
+        // Token keys
+        // https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html
+        uint256 p = uint256(slot);
+        storeUint(address(this), p, tokenInfo.token.tokenKeys.length);
+        uint256 itemSlot = uint256(keccak256(abi.encodePacked(p)));
+
+        for (uint256 i = 0; i < tokenInfo.token.tokenKeys.length; i++) {
+            // Key type
+            storeUint(address(this), itemSlot, tokenInfo.token.tokenKeys[i].keyType);
+            itemSlot += 1;
+
+            storeAddress(address(this), itemSlot, tokenInfo.token.tokenKeys[i].key.contractId);
+            itemSlot += 1;
+
+            storeBytes(address(this), itemSlot, tokenInfo.token.tokenKeys[i].key.ed25519);
+            itemSlot += 1;
+
+            storeBool(address(this), itemSlot, tokenInfo.token.tokenKeys[i].key.inheritAccountKey);
+            itemSlot += 1;
+
+            storeBytes(address(this), itemSlot, tokenInfo.token.tokenKeys[i].key.ECDSA_secp256k1);
+            itemSlot += 1;
+
+            storeAddress(address(this), itemSlot, tokenInfo.token.tokenKeys[i].key.delegatableContractId);
+            itemSlot += 1;
+        }
+        slot = bytes32(uint256(slot) + 1);
+
+        // Expiry
+        storeInt(address(this), uint256(slot), tokenInfo.token.expiry.second);
+        slot = bytes32(uint256(slot) + 1);
+
+        storeAddress(address(this), uint256(slot), tokenInfo.token.expiry.autoRenewAccount);
+        slot = bytes32(uint256(slot) + 1);
+
+        storeInt(address(this), uint256(slot), tokenInfo.token.expiry.autoRenewPeriod);
+        slot = bytes32(uint256(slot) + 1);
+
+        // Total supply
+        storeInt(address(this), uint256(slot), tokenInfo.totalSupply);
+        slot = bytes32(uint256(slot) + 1);
+
+        // Deleted
+        storeBool(address(this), uint256(slot), tokenInfo.deleted);
+        slot = bytes32(uint256(slot) + 1);
+
+        // Fixed fees
+        // https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html
+        p = uint256(slot);
+        itemSlot = uint256(keccak256(abi.encodePacked(p)));
+        storeUint(address(this), p, tokenInfo.fixedFees.length);
+        for (uint256 i = 0; i < tokenInfo.fixedFees.length; i++) {
+            storeInt(address(this), itemSlot, tokenInfo.fixedFees[i].amount);
+            itemSlot += 1;
+
+            storeAddress(address(this), itemSlot, tokenInfo.fixedFees[i].tokenId);
+            offset = 20;
```

Comment:
I think we shouldn't support `offset`s. This adds unnecessary complexity to both Foundry's and Hardhat's solution. I guess it's just easier to add _gaps_ into the struct.